test/evergreen.yml

```diff
@@ -5553,6 +5559,59 @@ buildvariants:
   tasks:
     - name: examples-c-tsan
 
+# This variant runs all possible tasks under TSan but instructs TSan to not report any issues it
+# encounters. This allows us to check whether a TSan change may cause other unexpected failures
+# while also not having to fix every TSan issue discovered. At the conclusion of the TSan work this
+# variant will be deleted.
+- name: ubuntu2004-tsan-no-throw
+  display_name: "~ Ubuntu 20.04 TSAN (No throw)"
+  patch_only: true
+  run_on:
+  # Defaulting to ubuntu2004-large. We can move back to ubuntu2004-test as we ID which tests can fit on the smaller host
+  - ubuntu2004-large
+  expansions:
+    CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_stable_clang.cmake
+    CMAKE_BUILD_TYPE: -DCMAKE_BUILD_TYPE=TSan
+    ENABLE_TCMALLOC: 0
+    data_validation_stress_test_args: -t r -m -W 3 -D -p -n 100000 -k 100000 -C cache_size=100MB
+    additional_env_vars: |
+      # We intentionally suppress all TSan warnings.
+      export TSAN_OPTIONS="detect_deadlocks=0:report_bugs=0:report_atomic_races=0:allocator_may_return_null=1"
+    num_jobs: $(echo "`grep -c ^processor /proc/cpuinfo`" | bc)
+    # We don't run C++ memory sanitized testing as it creates false positives. TSan also significantly increases the runtime
+    # of some csuite tests, so disable the already long tests as well as tests heavily impacted by TSan.
+    ctest_extra_args: -LE "cppsuite|long_running|sanitizer_long"
+    # TSan slows down python testing enough that we start printing "operation took X seconds" warnings.
+    # These don't impact functionality, but the python tests fail on detection of any content in stdout so ignore it instead.
+    ignore_stdout: --ignore-stdout
+    # Reduce the number of test iterations in validation stressing as TSan increases runtime by a factor of roughly 10x
+    validation_stress_num_runs: 20
+  tasks:
+  # The pull request tag, which includes azure gcp, doesn't compile under clang
+  #  - name: ".pull_request"
```

Comment:
You can exclude the azure and gcp tests for example:
```
".pull_request !azure-gcp-tiered-test-small"
```

Reply:
Updated in [b204cc2](https://github.com/wiredtiger/wiredtiger/pull/10839/commits/b204cc2e42fd2333fae0b6da7d6aef21bdd377ae)